### DISCUSSION TITLE
Nuxt3 retrieval from database is not displayed in source code (SEO).

### DISCUSSION THREAD

**Discussion by mydidefix** [nuxt/nuxt#16312] [Questions]
Nuxt3 retrieval from database is not displayed in source code (SEO). But the browser displays the data. Is that at all possible with nuxt that data from database is modified for (SEO) SSR.
Thanks for the help

**Comment by mirumirumi** [comment]
I am not sure what you are talking about, but it is a very common use case to get data from a database on one of the servers and then use it in Nuxt 3 to display it in the browser.

In other words, you can take the data in the database, process it for SEO, and display it in any format you want.

Did this help you?

**Reply by mydidefix** [reply]
Thanks for the feedback. Unfortunately the info does not help me.
So for Google to index the content, for example, the source code must show the content. This makes nuxt3 with me static content, there everything is ok. But if the content comes from the database then it is in the source code !--[-->. 
But the content should be displayed for example First name: Max Last name: Mustermann. Then Google can include the content in its search index.
Thanks for the help.

**Reply by Flanker72** [reply]
Seems that you are not awaiting that data. Nuxt with SSR mode by default builds the response and sends it to browser fully rendered.
You need to use `await useFetch(...)` or `await useAsyncData(...)` in your page/component to preload such data. Then it will become available to Nuxt and it will render all data before passing the result to server.

**Comment by TheAlexLichter** [comment]
Without a reproduction it will be tough to help you with this one sadly. Nuxt is capable of doing that in general, so please add some small example CodeSandbox or Stackblitz 😋

**Comment by JiatLn** [comment]
hi, could you tell me how to solve this problem?
